comparative_benchmark/xla_jax/run_benchmarks.py

```diff
@@ -0,0 +1,259 @@
+      "framework": str(model.model_impl.framework_type),
+      "data_type": data_type,
+      "batch_size": batch_size,
+      "inputs": input_dims,
+      "outputs": output_dims,
+      "compiler": "xla",
+      "device": benchmark.target_device.name,
+      "tags": model.model_impl.tags + model.tags,
+  }
+
+  print(f"\n\n--- {benchmark.name} ---")
+
+  # Retrieve framework-level benchmarks.
+  with multiprocessing.Manager() as manager:
+    shared_dict = manager.dict()
+
+    backend = benchmark.target_device.accelerator_type
+    kwargs: Dict[str, Any] = dict(
+        model=model,
+        warmup_iterations=warmup_iterations,
+        benchmark_iterations=iterations,
+        backend=backend,
+        shared_dict=shared_dict,
+    )
+    if run_in_process:
+      _run_framework_benchmark(**kwargs)
+    else:
+      p = multiprocessing.Process(target=_run_framework_benchmark,
+                                  kwargs=kwargs)
+      p.start()
+      p.join()
+
+    framework_metrics = dict(shared_dict)
+
+  return BenchmarkResult(
+      definition=benchmark_definition,
+      metrics={
+          "framework_level": framework_metrics,
+      },
+  )
+
+
+def _parse_arguments() -> argparse.Namespace:
+  parser = argparse.ArgumentParser(
+      description="Run JAX benchmarks with XLA backend.")
+  parser.add_argument(
+      "-o",
+      "--output_path",
+      type=pathlib.Path,
+      required=True,
+      help=
+      "Path to results json file. Expects this file to have been pre-populated."
+  )
+  parser.add_argument("-name",
+                      "--benchmark_name",
+                      type=str,
+                      required=True,
+                      help="The regex pattern to match benchmark names.")
+  parser.add_argument("-w",
+                      "--warmup_iterations",
+                      type=int,
+                      default=5,
+                      help="The number of warmup steps.")
+  parser.add_argument("-iter",
+                      "--iterations",
+                      type=int,
+                      default=100,
+                      help="The number of iterations to benchmark.")
+  parser.add_argument(
+      "--run_in_process",
+      action="store_true",
+      help=("Whether to run the benchmark under the same process. Set this to"
+            " true when profiling a single workload."))
+  parser.add_argument("--verbose",
+                      action="store_true",
+                      help="Show verbose messages.")
+
+  return parser.parse_args()
+
+
+def main(
+    benchmark_name: str,
+    run_in_process: bool,
+    warmup_iterations: int,
+    iterations: int,
+    output_path: pathlib.Path,
+    verbose: bool,
+):
+  name_pattern = re.compile(f"^{benchmark_name}$")
+  benchmarks = [
+      benchmark for benchmark in benchmark_definitions.BENCHMARKS
+      if name_pattern.match(benchmark.name)
+  ]
+
+  if not benchmarks:
+    raise ValueError(f'No benchmark matches "{benchmark_name}".')
```

Comment:
Might be useful to print out the list of available benchmarks.

Reply:
Done.